crates/dojo-bindgen/src/plugins/typescript_v2/mod.rs

```diff
@@ -0,0 +1,577 @@
+    // Handled tokens should be a list of all structs and enums used by the contract
+    // Such as a set of referenced tokens from a model
+    fn format_system(system: &Function, handled_tokens: &[Composite]) -> String {
+        let args = system
+            .inputs
+            .iter()
+            .map(|arg| {
+                format!(
+                    "{}: {}",
+                    arg.0,
+                    if TypeScriptV2Plugin::map_type(&arg.1.type_name()) == arg.1.type_name() {
+                        arg.1.type_name()
+                    } else {
+                        TypeScriptV2Plugin::map_type(&arg.1.type_name())
+                    }
+                )
+            })
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        let calldata = system
+            .inputs
+            .iter()
+            .map(|arg| {
+                let token = &arg.1;
+                let type_name = &arg.0;
+
+                match handled_tokens.iter().find(|t| t.type_name() == token.type_name()) {
+                    Some(t) => {
+                        // Need to flatten the struct members.
+                        match t.r#type {
+                            CompositeType::Struct => t
+                                .inners
+                                .iter()
+                                .map(|field| format!("props.{}.{}", type_name, field.name))
+                                .collect::<Vec<String>>()
+                                .join(",\n                    "),
+                            _ => type_name.to_string(),
+                        }
+                    }
+                    None => type_name.to_string(),
+                }
+            })
+            .collect::<Vec<String>>()
+            .join(",\n                ");
+
+        format!(
+            "async {pretty_system_name}({args}): Promise<void> {{
+        try {{
+            await this.execute(\"{system_name}\", [{calldata}])
+        }} catch (error) {{
+            console.error(\"Error executing {pretty_system_name}:\", error);
+            throw error;
+        }}
+    }}",
+            pretty_system_name = system.name.to_case(convert_case::Case::Camel),
+            // formatted args to use our mapped types
+            args = args,
+            system_name = system.name,
+            // calldata for execute
+            calldata = calldata
+        )
+    }
+
+    // Formats a contract file path into a pretty contract name
+    // eg. dojo_examples::actions::actions.json -> Actions
+    fn formatted_contract_name(contract_file_name: &str) -> String {
+        let contract_name =
+            contract_file_name.split("::").last().unwrap().trim_end_matches(".json");
+        contract_name.to_string()
+    }
+}
+
+#[async_trait]
+impl BuiltinPlugin for TypeScriptV2Plugin {
+    async fn generate_code(&self, data: &DojoData) -> BindgenResult<HashMap<PathBuf, Vec<u8>>> {
```

Comment:
For the tests, you should target this function.

`DojoData` can be loaded manually to test some edge cases, or you could load the spawn-and-move example into your test to generate the `DojoData` automatically using the `gather_dojo_data` function:
https://github.com/dojoengine/dojo/blob/f4c47ba7a0137e376556d3d7459f6b2569416dc4/crates/dojo-bindgen/src/lib.rs#L81

Then, you can test the output to be written into a file, and compare to your test file.

Reply:
Hmm, I'm still not fully seeing how to tackle this.

Won't using the spawn-and-move example require me to build the project first? Since it expects the `manifest_path`.

Or are there mocks available for this?